### PR TITLE
added common unit test use cases

### PR DESCRIPTION
**Pull request by guyfreund**
## Status
Ready

## Description
Added common unit test use cases we usually use when writing integrations/scripts

### CODE REVIEW

**Comment by guyfreund** [comment]
@jochman please add your use cases.

File: docs/howtos/integrations/unit-testing.md [hunk 1]
```diff
@@ -36,11 +36,132 @@ if __name__ == "__builtin__" or __name__ == "builtins":
...
+```
+from mock import patch
+```
+Next we can mock the __init__ method of ComplexObject once and then test all cases with parametrize, as follows:
+```
+@pytest.mark.parametrize('b, output', [([1], 1), ('B', 'b'), ({}, True)])
+def test_method_b(b, output):
+    with patch.object(ComplexObject, '__init__', lambda a, b, c, d, e, f: None):
+        complex_object = ComplexObject(None, None, None, None, None)
+        complex.object.b = b
```
Comment (ShahafBenYakir) [inline docs/howtos/integrations/unit-testing.md]:
```suggestion
        complex_object.b = b
```

File: docs/howtos/integrations/unit-testing.md [hunk 2]
```diff
@@ -36,11 +36,132 @@ if __name__ == "__builtin__" or __name__ == "builtins":
...
+```
+The correct way to test method_b is patching the ComplexObject object. First, in your test file import the patch method using the following line of code:
+```
+from mock import patch
+```
+Next we can mock the __init__ method of ComplexObject once and then test all cases with parametrize, as follows:
+```
+@pytest.mark.parametrize('b, output', [([1], 1), ('B', 'b'), ({}, True)])
+def test_method_b(b, output):
+    with patch.object(ComplexObject, '__init__', lambda a, b, c, d, e, f: None):
```
Comment (ShahafBenYakir) [inline docs/howtos/integrations/unit-testing.md]: Is `a` supposed to be `self`? if so let's change it to self

Comment (guyfreund) [reply docs/howtos/integrations/unit-testing.md]: Nope, should be random variable name, we are mocking the init function.

**Comment by guyfreund** [comment]
> @guyfreund I did some modifications. Also removed the Complex object part. Don't really agree with what was done there and I think it just adds confusion.

@glicht take a look at https://github.com/demisto/demisto-sdk/blob/3e042d7fb997d4df8f430bdf2558c63e200e9cd3/demisto_sdk/commands/split_yml/tests/extractor_test.py , I think it's crucial to use the object patching and not to initiate the extractor object each time we want to change an attribute.

**Comment by glicht** [comment]
@guyfreund let's discuss face to face. For now I am merging this as you added some good stuff.

**Pull request merged**